projects/storefrontlib/src/shared/services/trigger/config/trigger.service.ts

```diff
@@ -1,28 +1,61 @@
-import { Injectable, ComponentFactory } from '@angular/core';
+import {
+  Injectable,
+  ComponentFactory,
+  ComponentFactoryResolver,
+} from '@angular/core';
 import {
   TriggerOutletMapping,
   TriggerInlineMapping,
   TriggerUrlMapping,
+  TriggerConfig,
+  TRIGGER_CALLER,
+  TriggerMapping,
 } from './trigger-config';
 import { OutletService } from '../../../../cms-structure/outlet/index';
 
 @Injectable({ providedIn: 'root' })
 export class TriggerService {
-  constructor(protected outletService: OutletService<ComponentFactory<any>>) {}
+  // Keep a list of rendered elements
+  protected renderedCallers: TRIGGER_CALLER[] = [];
 
-  renderDialog(
-    config: TriggerOutletMapping | TriggerInlineMapping | TriggerUrlMapping,
-    template?: ComponentFactory<any>
-  ): void | string {
+  constructor(
+    protected outletService: OutletService<ComponentFactory<any>>,
+    protected triggerConfig: TriggerConfig,
+    protected componentFactoryResolver: ComponentFactoryResolver
+  ) {}
+
+  renderDialog(caller: TRIGGER_CALLER, component?: any): void | string {
```

Comment:
The return statement surprises me. Not sure what you're getting at. I would say there's nothing to return here. 
If we implement  the url/routing, we can bridge this to the RoutingService, taking a cxRoute or actual URL into account. Another reason to break up this service into strategies, so that the `RoutingRenderStrategy` could use the `RoutingService` as a dependency.